src/main/java/org/inferred/freebuilder/processor/PrimitiveOptionalProperty.java

```diff
@@ -0,0 +1,240 @@
+package org.inferred.freebuilder.processor;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.CaseFormat;
+import com.google.common.base.Optional;
+import org.inferred.freebuilder.processor.util.*;
+
+import javax.lang.model.type.TypeMirror;
+import java.util.OptionalDouble;
+import java.util.OptionalInt;
+import java.util.OptionalLong;
+
+import static org.inferred.freebuilder.processor.BuilderMethods.*;
+import static org.inferred.freebuilder.processor.util.Block.methodBody;
+
+/**
+ * This property class handles the primitive optional fields, including
+ * {@link OptionalDouble}, {@link OptionalLong}, and {@link OptionalInt}.
+ */
+public class PrimitiveOptionalProperty extends PropertyCodeGenerator {
+    @VisibleForTesting
+    enum OptionalType {
+        INT(OptionalInt.class, "int"),
```

Comment:
Unfortunately, since FreeBuilder still supports Java 1.6, we can't refer directly to J8 types in the code (though you can in the unit tests). Use QualifiedName.of("java.util", "OptionalInt") instead.

Reply:
I based my branch off of v2. Didn't you remove support for 1.6 and 1.7?

Reply:
Oh! Ha. Missed that. That branch is not compete yet, but as long as you're happy to wait, then cool.